Firestore/core/src/firebase/firestore/model/document_set.h

```diff
@@ -42,26 +42,11 @@ namespace firebase {
 namespace firestore {
 namespace model {
 
-/**
- * A C++ comparator that returns less-than, implemented by delegating to
- * an NSComparator.
- */
-class DocumentSetComparator {
+class DocumentComparator : public util::FunctionComparator<FSTDocument*> {
  public:
-  explicit DocumentSetComparator(NSComparator delegate = nil)
-      : delegate_(delegate) {
-  }
-
-  util::ComparisonResult Compare(FSTDocument* lhs, FSTDocument* rhs) const {
-    return util::MakeComparisonResult(delegate_(lhs, rhs));
-  }
-
-  bool operator()(FSTDocument* lhs, FSTDocument* rhs) const {
-    return util::Ascending(Compare(lhs, rhs));
-  }
+  using FunctionComparator<FSTDocument*>::FunctionComparator;
```

Comment:
Is this `using` doing something?  We don't seem to use `FunctionComparator` here...

Reply:
This declares that this class has the same constructors as its parent class without repeating them all here.